### PR TITLE
DS: Build Link

### PR DESCRIPTION
**Pull request by nearbycoder**
Adds the link component to the design system with the proper design specs from https://www.figma.com/file/5Kra5sXVMkAjM62Gh8DpM8/Draft-Documentation?node-id=201%3A11

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -4,6 +4,18 @@ import theme from '../lib/theme';
...
   actions: { argTypesRegex: '^on[A-Z].*' },
+  backgrounds: {
+    values: [
+      {
+        name: 'light',
+        value: '#ffffff',
+      },
+      {
+        name: 'dark',
+        value: '#182042',
```
Comment (nearbycoder) [inline .storybook/preview.js:15]: Updated to match the dark design of our figma files.

File: lib/components/link/LinkText.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { Box, useStyleConfig } from '@chakra-ui/react';
+
+export function LinkText({ variant, size, children, ...rest }) {
```
Comment (nearbycoder) [inline lib/components/link/LinkText.js:4]: Very simple component that just wraps the text for providing the styles inside the anchor tag.

File: lib/components/link/link.theme.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export const Link = {
+  baseStyle: {
+    display: 'inline-block',
```
Comment (nearbycoder) [inline lib/components/link/link.theme.js:3]: Needed to update some of the defaults to prevent issues with default boxShadow and borders.

**Review by nearbycoder — COMMENTED** [review]
@caseymhunt good for initial overview

**Comment by nearbycoder** [comment]
relevant storybook https://607731addb01d30021caeac2-xqjhcdnowf.chromatic.com/?path=/docs/link--default

File: lib/components/link/linktext.theme.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import { colors } from '../../theme/colors.theme';
+import * as font from '../../theme/font.theme';
+
+export const LinkText = {
+  baseStyle: {
+    display: 'inline-block',
+  },
+  sizes: {
+    5: {
+      fontSize: `${font.fontSizes[5]}`,
```
Comment (caseymhunt) [inline lib/components/link/linktext.theme.js:10]: 😆 You know the drill. Theme-aware syntax weeeee

**Review by caseymhunt — CHANGES_REQUESTED** [review]
🥳  Minor nitpicks:
* adjust theme aware syntax
* ~~consider renaming stories per @jaxatto 's comment in Chromatic. ~~

**Review by caseymhunt — APPROVED** [review]

**Pull request merged**